.github/actions/lint/action.yml

```diff
@@ -0,0 +1,67 @@
+name: 'Lint'
+description: 'Run linting checks'
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/checkout@v4
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Install & cache node_modules
+      uses: Khan/actions@shared-node-cache-v2
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Get All Changed Files
+      uses: Khan/actions@get-changed-files-v2
+      id: changed
+
+    - id: js-ts-files
+      name: Find .js, .ts changed files
+      uses: Khan/actions@filter-files-v1
+      with:
+        changed-files: ${{ steps.changed.outputs.files }}
+        extensions: '.js,.jsx,.ts,.tsx'
+
+    - id: eslint-reset
+      uses: Khan/actions@filter-files-v1
+      name: Files that would trigger a full eslint run
+      with:
+        changed-files: ${{ steps.changed.outputs.files }}
+        files: '.eslintrc.js,yarn.lock,.eslintignore'
+
+    - id: typecheck-reset
+      uses: Khan/actions@filter-files-v1
+      name: Files that would trigger a typecheck run
+      with:
+        changed-files: ${{ steps.changed.outputs.files }}
+        files: '.yarn.lock'
+
+    # Linting / type checking
+    - name: Eslint
+      uses: Khan/actions@full-or-limited-v0
+      with:
+        full-trigger: ${{ steps.eslint-reset.outputs.filtered }}
+        full: yarn lint:ci .
+        limited-trigger: ${{ steps.js-ts-files.outputs.filtered }}
+        limited: yarn lint:ci {}
+
+    - name: Typecheck
+      if: always() # always run this check until we update the eslint config
+      # if: steps.js-ts-files.outputs.filtered != '[]' || steps.typecheck-reset.outputs.filtered != '[]'
+      shell: bash
```

Comment:
The `action.yml` files contain the same steps as before with the addition of `shell: bash` on the steps that had errors around requiring the `shell` config